README.md

```diff
@@ -93,23 +117,134 @@ externalFilters?:  boolean;
 externalFiltersState?:  ColumnFiltersState;
 
 };
+
 ```
 
 **columns:** an Array of ColumnDef<T> that specify the columns with all the props needed (see Tanstack docs)
+
 **useQueryHook:** The query hook generated by your Redux Toolkit Query API for fetching table data that should return an object like this
 
-    data: {
-    	data:  retData.data,
-    	count:  retData.count,
-    	page:  page,
-    	limit:  pageSize,
-    }
+data: {
+
+data: retData.data,
+
+count: retData.count,
+
+page: page,
+
+limit: pageSize,
+
+}
 
 **isDraggable:** Enables column sorting trough drag & drop
+
 **disableHeaderFilters:** this remove all the column headers filters (if you dont want to specify all _false_ in the **columns** array
+
 **externalFilters:** this enable the use of external filters (external means that filters are not in the columns header)
+
 **externalFiltersState:** _ColumnFiltersState_ object with the state of external filters (this is mandatory if **externalFilters** is true
 
+## Authentication
+
+This project is implemented with a flexible authentication provider that can support multiple providers.
+Now it support **MSAL** and **AUTH0** providers and you can switch between one another easily.
+This is how:
+
+1.  Go to the **index.tsx** file
+2.  Instantiate the implementation of your choice of AuthProvider interface
+
+```Typescript
+const  authProvider  =  new  Auth0AuthProvider(env);
+```
+
+in this case we choose the **Auth0** implementation.
+_env_ is the enviroment and it must contains all the data needed to authenticate with the provider.
+For this reason make sure that the **connectionStrings.js** file is aligned with the deploy environments you are using.
+In order to make this works locally you must create a **.env.local** file in the root of your project with all the env variables needed by connectionStrings.js and your AuthProvider implementation.
+example for **Auth0** :
+
+### .env.local
+
+```javascript
```

Comment:
this shuold go in a section dedicated on how to configure Auth0.
the same is needed for MSAL

Reply:
this wanted to be just an example, or we have to implement the readme for each authentication provide we may need and also update them when a change occour

Reply:
either the Providers use the config object from the 3rd party library or, like seems you've done, you define a different Config to be passed to the ctor of the Provider, then yes documentation is needed.